docs/_posts/examples/3400-01-14-filter-features-within-map-view.html

```diff
@@ -0,0 +1,229 @@
+    }
+}
+
+function normalize(string) {
+    return string.trim().toLowerCase();
+}
+
+function setFilter(features) {
+    map.setFilter('airport', ['in', 'abbrev'].concat(features));
+}
+
+map.on('load', function() {
+
+    // Add our vector tile source: World wide
+    // airports provided by Natural Earth
+    map.addSource('airports', {
+        "type": "vector",
+        "url": "mapbox://mapbox.04w69w5j"
+    });
+
+    map.addLayer({
+        "id": "airport",
+        "source": "airports",
+        "source-layer": "ne_10m_airports",
+        "type": "symbol",
+        "layout": {
+            "icon-image": "airport-15",
+            "icon-padding": 0
+        },
+        "filter": ["in", "abbrev", ""]
+    });
+
+    map.on('moveend', function() {
+        var features = map.querySourceFeatures('airports', {
+            sourceLayer: 'ne_10m_airports'
+        });
+
+        if (features) {
+            // Populate features for the listing overlay.
+            renderFeatures(features);
+
+            // Clear the input container
+            filter.value = '';
+
+            // Store the current features in sn `airports` variable to
+            // later use for filtering on `keyup`.
+            airports = features;
+
+            // Set the filter to populate features into the layer.
+            setFilter(features.map(function(feature) {
+                return feature.properties.abbrev;
+            }));
+        }
+    });
+
+    map.on('mousemove', function(e) {
+        var features = map.queryRenderedFeatures(e.point, {
+            layers: ['airport']
+        });
+
+        // Change the cursor style as a UI indicator.
+        map.getCanvas().style.cursor = features.length ? 'pointer' : '';
+
+        if (!features.length) {
+            popup.remove();
+            return;
+        }
+
+        var feature = features[0];
+
+        // Populate the popup and set its coordinates
+        // based on the feature found.
+        popup.setLngLat(feature.geometry.coordinates)
+            .setText(feature.properties.name + ' (' + feature.properties.abbrev + ')')
+            .addTo(map);
+    });
+
+    filter.addEventListener('keyup', function(e) {
+        var value = normalize(e.target.value);
+
+        // Filter visible features that don't match the input value.
+        var filtered = airports.filter(function(feature) {
+            var name = normalize(feature.properties.name);
+            var code = normalize(feature.properties.abbrev);
+            return name.indexOf(value) > -1 || code.indexOf(value) > -1;
+        });
+
+        // Populate the sidebar with filtered results
+        renderFeatures(filtered);
+
+        // Set the filter to populate features into the layer.
+        setFilter(filtered.map(function(feature) {
+            return feature.properties.abbrev;
+        }));
+    });
+
+    // Call this function on initialization
+    // passing an empty array to render an empty state
+    renderFeatures([]);
```

Comment:
Can we change this line to 

```js
renderFeatures(map.queryRenderedFeatures(e.point, {
  layers: ['airport']
}));
```

and get rid of "Drag the map to populate results" mode?